ambry-clustermap/src/test/java/com.github.ambry.clustermap/MockDataNodeId.java

```diff
@@ -56,6 +51,32 @@ private void populatePorts(ArrayList<Port> ports) {
     }
   }
 
+  /**
+   * Set the datacenters to which ssl is enabled. If the datacenter on which this datanode resides is
+   * part of the datacenters to which ssl is enabled, then this datanode will always return an SSL port in
```

Comment:
customizing is hard when the same clustermap is used by all nodes. The clustermap gets initialized with a config and that config has to be different for the clustermap of each node. For the main code, this makes things cleaner. In tests it would be cumbersome to create too many clustermaps unfortunately.